.github/workflows/cloud_build_failure_reporter.yml

```diff
@@ -0,0 +1,151 @@
+
+                  // updateIssues comments on any existing issues. No-op if no issue exists.
+                  const updateIssues = async function (txt) {
+                    if (prevIssues.length < 1) {
+                      console.log('no previous issues found.');
+                      return;
+                    }
+                    for (const prevIssue of prevIssues) {
+                      console.log(`found previous issue ${prevIssue.html_url}, adding comment`);
+                      await github.rest.issues.createComment({
+                        ...context.repo,
+                        issue_number: prevIssue.number,
+                        body: txt
+                      });
+                    }
+                  };
+
+                  // Find status of check runs.
+                  // We will find check runs for each commit and then filter for the periodic.
+                  // Checks API only allows for ref and if we use main there could be edge cases where
+                  // the check run happened on a SHA that is different from head.
+                  const commits = await github.paginate(github.rest.repos.listCommits, {
+                    ...context.repo
+                  });
+
+                  var foundCheck = false;
+                  let periodicCheck = {};
+
+                  for (const commit of commits) {
+                    console.log(
+                      `checking runs at ${commit.html_url}: ${commit.commit.message}`
+                    );
+                    const checks = await github.rest.checks.listForRef({
+                      ...context.repo,
+                      ref: commit.sha
+                    });
+                    // find runs for this commit
+                    for (const check of checks.data.check_runs) {
+                      console.log(`found run ${check.name} for ${commit.html_url}`);
+                      if (check.name.includes('nightly')) {
+                        foundCheck = true;
+                        periodicCheck = check;
+                        break;
+                      }
+                    }
+
+                    if (foundCheck) {
+                      if (
+                        periodicCheck.status === 'completed' &&
+                        periodicCheck.conclusion === 'success'
+                      ) {
+                        updateIssues(
+                          `[Passing periodic](${periodicCheck.html_url}) at ${commit.html_url}. Closing this issue.`
+                        );
+                      } else if (periodicCheck.status === 'in_progress') {
+                        console.log(
+                          `Check is pending ${periodicCheck.html_url} for ${commit.html_url}. Retry again later.`
+                        );
+                      }
+                      // error case
+                      else {
+                        createOrCommentIssue(
+                          'Failing periodic',
```

Comment:
nit: what message do we want maintainers to see? Should we provide a better comment/github issue (body and title) message with how this bug should be handled?

Reply:
Improved message.